Respond to duke feedback

C-Loftus opened the pull request:
# Description

- Gets rid of the fetched markdown in the footer. This is not the central source of truth, the docs website is, and it refers to Github info that isn't relevant.  
- Adds more detail to the CSV formatting reference and has it open by default. 
    - For more detail, I am still linking to the docs. I think it is better to link to the docs rather than include big blocks of text that duplicate info we have on the docs site. 
- Added an additional details admonition block that calls out where to find specific useful info in the docs

C-Loftus committed:
respond to duke feedback

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

ksonda commented:
Suggest something aboiut how what you put in the namespace will be created if it does not already exist, and your identifiers will just be added to the existing namespace if it already exists. And maybe slightly more explicit language that 

1. Identifiers are meant for individual monitoring locations
2. Identifiers are meant to redirect to a unique, individual page for each monitroing location

C-Loftus committed:
respond to feedback

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

C-Loftus commented:
Responded to the feedback. 
- Added addition info for the Identifiers. Wasn't sure where was best, but put it under the `id` description in the CSV info section.
- Added an italic callout for the fact that the namespace will be either created or updated
- Added an h2 element above the readme upload to specify it is either add or update


![image](https://github.com/user-attachments/assets/ed1f3dc5-c86b-4549-a0e2-9cef55cd4d09)

webb-ben commented:
Remove marked from package.json please

C-Loftus committed:
remove marked

webb-ben commented:
The page about the namespace creation should be a separate "step" from the uploading of the CSV. I suggest the implementation of [stepper](https://vuetifyjs.com/en/components/steppers/#usage) to better address the namespace specific requirements. Perhaps this is a first step where you either select the namespace from GitHub API populated list, or create a new one. 

Second step is uploading and formatting the CSV file

C-Loftus committed:
refactor to use stepper. Make code cleaner by separating into separate components

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

C-Loftus commented:
@webb-ben when you get a moment would you mind letting me know what you think of the v-stepper change?

It is not done yet, still have to make the validation code work in each tab and improve the submission tab.  (Wanted to check with you before I went in too deep or if you want anything else)

Thanks for recommending to use v-stepper. It looks nice and is a great way to split up the form semantically. (Also allows us to add extra background info without cramming the form)

C-Loftus committed:
work on colorscheme; refactoring; and better errors

C-Loftus committed:
fix typecheck

C-Loftus commented on src/components/MetadataGenerator.vue:
```diff
@@ -0,0 +1,112 @@
+      return this.namespace && this.namespace.length > 0 && this.error.length === 0
+    },
+
+    readme() {
+      if (!this.readmeAlreadyUploaded) {
+        const requiredReadmeFields: MarkdownSection[] = [
+          { body: this.homepage, sectionName: 'Homepage' },
+          { body: this.description, sectionName: 'Description' },
+          { body: this.example_pid, sectionName: 'Example PID' },
+          { body: this.example_redirect_target, sectionName: 'Example redirect target URL' },
+          { body: this.contact_name, sectionName: 'Contact name' },
+          { body: this.contact_email, sectionName: 'Contact email' }
+        ]
+
+        for (const field of requiredReadmeFields) {
+          if (field.body.length == 0) {
+            this.error = `${field.sectionName} is required`
+            return null
+          }
+        }
+
```
I have this as a computed property, but it doesn't seem to be updating the error. Do you have any idea why this wouldn't be running on an update of the form fields? The logic is right, I used the same before the refactor. Something is off with how the computed property is being updated.

C-Loftus commented on src/components/PageHeader.vue:
```diff
@@ -1,37 +1,68 @@
+  /* Space out the logo and the theme icon */
+  padding: 10px;
+
+  background-color: var(--v-theme-primary);
```
This was how I was trying to refer to the primary colors from the theme. I can see this is a variable that is defined in the browser and I have a primary color in my theme but for some reason it is not working.

C-Loftus reviewed (commented):
Still more work to be done on this but switching off for a bit to work on some other stuff since I am getting a bit stuck. @webb-ben if you have a moment these were the two issues I was preferring to.

Otherwise I will take a look at thisagain later

C-Loftus committed:
update

C-Loftus committed:
catch when readme is null

C-Loftus commented:
Since we do not have a logo that works with a dark background, should I just remove the dark mode toggle? I think dark mode is a nice quality of life improvement, but that being said, I can't find a way to make the logos look good on a dark background without adding a white border. Not sure what is best

webb-ben commented:
> Since we do not have a logo that works with a dark background, should I just remove the dark mode toggle? I think dark mode is a nice quality of life improvement, but that being said, I can't find a way to make the logos look good on a dark background without adding a white border. Not sure what is best

I think we can implement it in a future sprint if desired. No reason to be hung up on it right now

C-Loftus committed:
working now and blocking user if invalid data is present

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

C-Loftus commented:
Should be ready to review: 

Things I added:
* Stepper component breaks up the form into logical parts
* Added some more verbiage/headers 
* Refactored code based to be more modular

Things I will add in a future PR:
* Started dark mode but we don't have logos for a dark theme for not a good fit at the moment
* Will refactor to use pinia for state management once the logic gets more complex

Things I wasn't sure about:
* Unclear how much information we want to put on the last submission tab (too much potentially confuses them, but not having enough makes it unclear where the data was actually submitted to). I could link to the GitHub pull request once it is submitted but once again not sure if this is helpful or adds more confusion.

C-Loftus committed:
fix minor wording issue

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci
